pages/instructorFileEditor/instructorFileEditorClient.js

```diff
@@ -1,22 +1,38 @@
 /* eslint-env browser,jquery */
 /* global ace */
-window.InstructorFileEditor = function(uuid, options) {
-    const elementId = '#file-editor-' + uuid;
-    this.element = $(elementId);
+/* global CryptoJS */
+window.InstructorFileEditor = function(options) {
+    this.element = $(`#${options.elementId}`);
     if (!this.element) {
-        throw new Error('Instructor file editor element ' + elementId + ' was not found!');
+        throw new Error(`Instructor file editor element ${options.elementId} was not found!`);
     }
 
+    this.origHash = options.origHash;
+    this.saveElement = $(`#${options.saveElementId}`);
     this.inputElement = this.element.find('input[name=file_edit_contents]');
     this.editorElement = this.element.find('.editor');
-    this.editor = ace.edit(this.editorElement.get(0));
-    this.editor.setTheme('ace/theme/chrome');
-    this.editor.getSession().setUseWrapMode(true);
-    this.editor.setShowPrintMargin(false);
-    this.editor.getSession().on('change', this.syncFileToHiddenInput.bind(this));
+    this.editor = ace.edit(this.editorElement.get(0), {
+        minLines: 10,
+        maxLines: Infinity,
+        autoScrollEditorIntoView: true,
+        wrap: true,
+        showPrintMargin: false,
+        mode: options.aceMode,
+        readOnly: options.readOnly,
+    });
 
-    if (options.aceMode) {
-        this.editor.getSession().setMode(options.aceMode);
+    if (options.altElementId) {
+        this.editor.setReadOnly(true);
+        this.altElement = $(`#${options.altElementId}`);
+        this.chooseElement = this.element.find('button[id=choose]');
```

Comment:
I took a quick look at the tests and it looked to me like we are testing the case above that didn’t work for me, so I’m not sure whether I just did something weird, or we don’t actually have a test for this?

Reply:
We don't have a test for this, because it's a client-side thing. In any case, the problem has now been addressed.